source/Xunit.Gherkin.Quick/CoreModel/Feature/Discovery/FeatureDiscoveryModel.cs

```diff
@@ -1,28 +1,60 @@
 ﻿using System;
-
+using System.IO;
+using System.Collections.Generic;
+using System.Linq;
 namespace Xunit.Gherkin.Quick
 {
     internal sealed class FeatureDiscoveryModel
     {
+        internal sealed class FeatureFile {
+            public string Path { get; }
+            public global::Gherkin.Ast.Feature Feature { get; }
+
+            internal FeatureFile(string path ,global::Gherkin.Ast.Feature feature) {
+                this.Path = path;
+                this.Feature = feature;                
+            }
+        }
+
         private readonly IFeatureFileRepository _featureFileRepository;
 
         public FeatureDiscoveryModel(IFeatureFileRepository featureFileRepository)
         {
             _featureFileRepository = featureFileRepository ?? throw new ArgumentNullException(nameof(featureFileRepository));
         }
 
-        public global::Gherkin.Ast.Feature Discover(Type featureClassType)
+        public List<FeatureFile> Discover(Type featureClassType)
         {
             if (featureClassType == null)
                 throw new ArgumentNullException(nameof(featureClassType));
 
-            var fileName = FeatureClassInfo.FromFeatureClassType(featureClassType).FeatureFilePath;
-            var featureFile = _featureFileRepository.GetByFilePath(fileName);
-            if (featureFile == null)
-                throw new System.IO.FileNotFoundException("Feature file not found.", fileName);
+            var fileClassInfo = FeatureClassInfo.FromFeatureClassType(featureClassType);
+
+            var featureFilePaths = new List<string>();
+                
+            if (fileClassInfo.IsPattern) {
+                featureFilePaths.AddRange(
+                    _featureFileRepository.GetFeatureFilePaths()
+                    .FindAll(f => fileClassInfo.MatchesFilePathPattern(f))
+                );
+            } else {
+                var fileName = fileClassInfo.FeatureFilePath;
+                var featureFile = _featureFileRepository.GetByFilePath(fileName);
+                if (featureFile == null)
+                    throw new System.IO.FileNotFoundException("Feature file not found.", fileName);
+
+                featureFilePaths.Add(fileName);
+            }
+
+            var newFeatures = featureFilePaths
+                .Select(f => new FeatureFile(f, _featureFileRepository.GetByFilePath(f).GherkinDocument.Feature))
+                .ToList() ?? new List<FeatureFile>();
+
+            if (newFeatures.Count == 0) {

```

Comment:
Move this inside the If statement above, right after featureFilePaths.AddRange, that will make more sense for the future code readers. Also, no computation above will be done if featureFilePaths.Count==0.